### PR TITLE
Bszabo/frontend app authoring pipeline testing

### PR DESCRIPTION
**Pull request by bszabo**
Tubular script ultimately intended to invoke Datadog synthetic tests in service of a GoCD build pipeline. This early version, however, does not do that. It only executes a single hack-y test intended to be used like a Django waffle switch to control the behavior of the GoCD pipeline responsible for running synthetic tests.

This tubular script expects GoCD configuration to provide keys for Datadog API use as environment variables. It also hard-wires the set of Datadog synthetic tests to be run, and currently has that list be empty.

The one failing unit test was merged into master following an earlier PR and needs to be resolved separately.

Testing on stage: Test in conjunction with the `edx-internal` [PR](https://github.com/edx/edx-internal/pull/12296) that creates a new stage-frontend-app-course-authoring pipeline to use of this script. Note that the new pipeline will succeed or fail depending on how the "waffle switch" test has been written. Course authoring builds will be unaffected by whether the new pipeline succeeds or fails.

### CODE REVIEW

File: tubular/scripts/dd_synthetic_tests.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        self.test_run_id = None     # The run ID given by Datadog to this test's invocation
+        self.success = None
+
+class DatadogClient:
+    ''' Client class to invoke datadog API to run and monitor synthetic tests '''
+
+    DATADOG_SYNTHETIC_TESTS_API_URL = "https://api.datadoghq.com/api/v1/synthetics/tests"
+    MAX_ALLOWABLE_TIME_SECS = 600 # 10 minutes
+
+    WAFFLE_SWITCH_TEST = SyntheticTest(
```
Comment (dianakhuang) [inline tubular/scripts/dd_synthetic_tests.py]: This is a bit of a nitpick, but since 'waffle' is the name of the specific library we use in Django to handle toggles, this name is a bit confusing since it doesn't test 'waffle' functionality at all. Maybe something like `TESTS_ENABLED`? (I am not sure I'm any better at naming 😅 .)

File: tubular/scripts/dd_synthetic_tests.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        self.name = name            # The test's Datadog name
+        self.public_id = public_id  # The test's Datadog Test ID
+        self.test_run_id = None     # The run ID given by Datadog to this test's invocation
+        self.success = None
+
+class DatadogClient:
+    ''' Client class to invoke datadog API to run and monitor synthetic tests '''
+
+    DATADOG_SYNTHETIC_TESTS_API_URL = "https://api.datadoghq.com/api/v1/synthetics/tests"
+    MAX_ALLOWABLE_TIME_SECS = 600 # 10 minutes
```
Comment (dianakhuang) [inline tubular/scripts/dd_synthetic_tests.py:24]: This is also a bit of a nitpick, but do we want to make this also configurable via the command line interface so that, in theory, different pipelines can use different timeouts?

**Review by dianakhuang — APPROVED** [review]
Overall, looks great. Just some comments/nitpicks that we can discuss if necessary.

File: tubular/scripts/dd_synthetic_tests.py (hunk 1 above)
Comment (bszabo) [reply tubular/scripts/dd_synthetic_tests.py:26]: I'll need to handle passing configuration settings via the CLI invocation in the next pass. Will include timeouts in that. Good call, thanks.

File: tubular/scripts/dd_synthetic_tests.py (hunk 2 above)
Comment (bszabo) [reply tubular/scripts/dd_synthetic_tests.py:24]: I'll be including the passing of configuration settings via the CLI in the next pass, and will include timeouts in that. Good call, thanks.

File: tubular/scripts/dd_synthetic_tests.py (hunk 1 above)
Comment (bszabo) [reply tubular/scripts/dd_synthetic_tests.py]: I'll rename to DEPLOYMENT_TESTING_ENABLE_DISABLE_SWITCH

Comment (timmc-edx) [reply tubular/scripts/dd_synthetic_tests.py]: Generally the format we'd use is something like `DEPLOYMENT_TESTING_ENABLED`. (Sometimes you also see `ENABLE_DEPLOYMENT_TESTING` but then settings don't sort very well when you're looking at an alphabetized listing!)

Comment (bszabo) [reply tubular/scripts/dd_synthetic_tests.py]: I've gone with a compromise, as this macro isn't for the boolean itself, but rather for the source of the boolean
